### PR TITLE
[FLO-1768] feat: Support running multiple users locally

### LINKED ISSUE #164: Run multiple users locally
Create a multi-headed Element runner that runs a user a unique tab, instead of a unique browser instance.

### PR DESCRIPTION
**Pull request by phileba**
As a user, I want to run multiple users locally

#Usage:
```
export const settings: TestSettings = {
	stages: [
		{
			duration: '1h',
			target: 2,
		},
	],
}
```
resolve #164

### CODE REVIEW

File: packages/core/api.ts [hunk 1]
```diff
@@ -33,3 +36,6 @@ export { StructuredError } from './src/utils/StructuredError'
 export { default as NetworkRecorder } from './src/network/Recorder'
 export { Entry } from './src/network/Protocol'
 export { default as NetworkObserver } from './src/runtime/test-observers/NetworkObserver'
+export { mustCompileFile } from './src/TestScript'
+export { DEFAULT_SETTINGS } from './src/runtime/Settings'
+export { normalizeElementOptions } from './src/ElementOption'
```
Comment (HieuLeKMS) [inline packages/core/api.ts:41]: is `./src/ElementOption` correct? I can't seem to find it

File: packages/core/src/Element.ts [hunk 2]
```diff
@@ -88,15 +86,16 @@ export async function runCommandLine(args: ElementRunArguments): Promise<void> {
 			order++
 			const fileTitle = chalk.grey(`${file} (${order} of ${numberOfFile})`)
 			console.group(chalk('Running', fileTitle))
-			const opts = normalizeElementOptions(arg, cache)
+			const opts: ElementOptions = normalizeElementOptions(arg, cache)
```
Comment (HieuLeKMS) [inline packages/core/src/Element.ts:89]: usually, the variable get its type from the return type of the function, in that case there's no need to specify its type

File: packages/core/src/runtime/test-observers/LifecycleObserver.ts [hunk 3]
```diff
@@ -30,13 +30,15 @@ export default class LifecycleObserver implements TestObserver {
 	}
 	async onStepPassed(test: Test, step: Step) {
 		const testObserver: TestObserver = this.next
-		await testObserver.afterStep(test, step)
+		//await testObserver.afterStep(test, step)
```
Comment (HieuLeKMS) [inline packages/core/src/runtime/test-observers/LifecycleObserver.ts:33]: 🔥

File: packages/core/src/runtime/test-observers/TimingObserver.ts [hunk 4]
```diff
@@ -125,37 +125,12 @@ export class TimingObserver extends NetworkRecordingTestObserver {
...
-
-			// let paintTimingEntries = await browser.paintTiming()
-			// paintTimingEntries.forEach(entry => {
-			// if (measurementKeysForDOM[entry.name]) {
-			// browserPerformanceTiming[measurementKeysForDOM[entry.name]] = Math.round(entry.startTime)
-			// }
-			// })
-
-			// reporter.addCompoundMeasurement('browser_performance', browserPerformanceTiming, name)
-
```
Comment (HieuLeKMS) [inline packages/core/src/runtime/test-observers/TimingObserver.ts:151]: 🔥

File: packages/core/src/runtime/test-observers/TimingObserver.ts [hunk 5]
```diff
@@ -125,37 +125,12 @@ export class TimingObserver extends NetworkRecordingTestObserver {
...
 			reporter.addMeasurement('throughput', networkRecorder.networkThroughput(), name)
 			reporter.addMeasurement('response_time', documentResponseTime, name)
 			reporter.addMeasurement('latency', documentLatency, name)
 		}
 
 		reporter.addMeasurement('transaction_rate', 1, name)
-		// reporter.addMeasurement('concurrency', 1, name) //TODO see whether concurrency changes in go grid work with this
```
Comment (HieuLeKMS) [inline packages/core/src/runtime/test-observers/TimingObserver.ts:158]: 🔥

File: packages/core/api.ts (hunk 1 above)
Comment (phileba) [reply packages/core/api.ts:41]: yep, it correct. Can you reload your IDE and check it again.

File: tsconfig.json [hunk 6]
```diff
@@ -48,7 +48,8 @@
 		"**/dist",
 		"*.spec.ts",
 		"*.test.ts",
-		"**/node_modules",
+		"**/__tests__/**",
+    	"**/node_modules",
```
Comment (HieuLeKMS) [inline tsconfig.json:52]:
```suggestion
    		"**/node_modules",
```

File: packages/report/src/types/CallSite.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
+export interface CallSite {
+	file?: string
+	code?: string
+	line: number
+	column: number
+	level?: number
+}
+
+export function callSiteToString(callSite: CallSite | undefined): string {
```
Comment (HieuLeKMS) [inline packages/report/src/types/CallSite.ts:9]: This is defined in `packages/core/src/test-script/SourceUnmapped.ts` as well. Is it intended?

File: examples/internet-herokuapp/test.ts [hunk 8]
```diff
@@ -0,0 +1 @@
+class Test {}
```
[the remark this answers is not in the document]
Comment (phileba) [reply examples/internet-herokuapp/test.ts:1]: this is the sample test file, for this one.
```
	step('Test: 03 - Choose File', async browser => {
		let fileInput = await browser.findElement('input[type="file"]')
		await fileInput.uploadFile('test.ts')
	})
```

File: packages/report/src/types/CallSite.ts (hunk 7 above)
Comment (phileba) [reply packages/report/src/types/CallSite.ts:9]: removed from `SourceUnmapped.ts`

File: tsconfig.json [hunk 9]
```diff
@@ -1,54 +1,72 @@
...
+    "packages/core/node_modules",
+    "packages/core/src/runtime/errors/ErrorClassification.ts",
+    "**/dist",
+    "*.spec.ts",
+    "*.test.ts",
+    "**/__tests__/**",
+    "**/node_modules",
+    "node_modules"
+  ]
+}
```
Comment (HieuLeKMS) [inline tsconfig.json:72]: missing new line

**Review by HieuLeKMS — APPROVED** [review]

**Pull request merged**